src/mpid/ch4/src/ch4_impl.h

```diff
@@ -997,8 +1005,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_compute_acc_op(void *source_buf, int source_
 
             MPIR_Assign_trunc(count, curr_len / type_size, int);
 
-            (*uop) ((char *) source_buf + src_type_stride * accumulated_count,
-                    (char *) target_buf + MPIR_Ptr_to_aint(curr_loc), &count, &type);
+            if (src_ptr) {
+                memcpy(src_ptr, (char *) source_buf + src_type_stride * accumulated_count,
```

Comment:
Shouldn't we use `MPIR_Typerep_unpack` here? I think this will fix the issues with accumulate on pairtypes with padding.

Reply:
`MPI_SHORT_INT` is currently xfailed because the accumulate ops do not account for the missing 2-byte pad in the packed version.

Reply:
> 
> 
> Shouldn't we use `MPIR_Typerep_unpack` here?

Are we concerned that `source_buf` may be GPU buffer (the reason you suggested `MPIR_Typerep_unpack`)?

Reply:
> Shouldn't we use `MPIR_Typerep_unpack` here? I think this will fix the issues with accumulate on pairtypes with padding.

Ahh, I was being dense. Finally I realized for pair types, it is not just omitting the padding during packing; it is literally *packed* as a non-contig data. Thus to unpack, we need use `MPIR_Typerep_unpack` -- for the sake of `MPI_SHORT_INT`.